tests/unit/backend/chalice/api_server/test_revisions.py

```diff
@@ -232,57 +231,86 @@ def test__delete_published_dataset_during_revision(self):
+        response = self.app.get(path, headers)
+        response.raise_for_status()
+        ids = [col["id"] for col in json.loads(response.body)["collections"]]
+        self.assertIn(collection_id, ids)
 
-    def get_s3_objects_from_collections(self) -> typing.Tuple[typing.List, typing.List]:
-        """
-        :return: a list of s3 objects in the published collection, and a list of s3 objects the revision collection.
-        """
-        rev_s3_objects = []
-        pub_s3_objects = []
+        # check get collection_uuid
+        path = f"/dp/v1/collections/{collection_id}"
+        headers = {"host": "localhost", "Content-Type": "application/json"}
+        response = self.app.get(path, headers)
+        response.raise_for_status()
+        actual = json.loads(response.body)
+        self.assertEqual("PUBLIC", actual["visibility"])
+        self.assertEqual(collection_id, actual["id"])
+        return actual
+
+    def verify_datasets(self, actual_body, expected_dataset_ids):
+        actual_datasets = [d["id"] for d in actual_body["datasets"]]
+        self.assertListEqual(expected_dataset_ids, actual_datasets)
+        self.assertTrue(all([d["published"] for d in actual_body["datasets"]]))
+        for dataset_id in expected_dataset_ids:
+            dataset = Dataset.get(self.session, dataset_id)
+            for s3_object in self.get_s3_object_paths_from_dataset(dataset):
+                if dataset.tombstone:
+                    self.assertS3FileDoesNotExist(*s3_object)
+                else:
+                    self.assertS3FileExists(*s3_object)
 
-        for i in range(len(self.pub_collection.datasets)):
-            pub_s3_objects.extend(self.get_s3_object_paths_from_dataset(self.pub_collection.datasets[i]))
-            rev_s3_objects.extend(self.get_s3_object_paths_from_dataset(self.rev_collection.datasets[i]))
-        return pub_s3_objects, rev_s3_objects
+    def test__publish_revision_with_collection_info_updated__201(self):
+        expected_body = {
+            "name": "collection name",
+            "description": "This is a test collection",
+            "contact_name": "person human",
+            "contact_email": "person@human.com",
+            "links": [
+                {"link_name": "DOI Link", "link_url": "http://doi.org/10.1016", "link_type": "DOI"},
+                {"link_name": "DOI Link 2", "link_url": "http://doi.org/10.1017", "link_type": "DOI"},
+            ],
+            "data_submission_policy_version": "0",
+        }
+        self.rev_collection.update(**expected_body)
+        pub_s3_objects, rev_s3_objects = self.get_s3_objects_from_collections()
+        actual_body = self.publish_collection(self.rev_collection.id)
+        self.assertPublishedCollectionOK(expected_body, pub_s3_objects)
+        self.verify_datasets(actual_body, [ds.id for ds in self.pub_collection.datasets])
+
+    def test_with_revision_and_existing_datasets(self):
```

Comment:
Do you want to explicitly test that a newly created or refreshed dataset isnt published?

Reply:
for that we need #1177, but I will add a test for that in that PR.

Reply:
this won't actually bock. I removed that part. I'll add it in a separate PR.